### DISCUSSION TITLE
Add friends bug

### DISCUSSION THREAD

**Discussion by ghost** [cssbattle/forum#82] [Bugs reports]
![image](https://user-images.githubusercontent.com/57074296/128537890-86a75fa4-cbcd-4fd7-a49e-86a6bb69d2ee.png)
if i trying to add my friend to friends his screen is black and its error here

**Comment by chinchang** [comment]
Hi @iSebek
Which page is this screenshot taken on? Also, were you logged in at that point?

**Comment by sylviaawuor** [comment]
This typically occurs when the code expects a certain data structure to be available, but instead, it receives 'null' or 'undefined'.
Add conditional check eg.

`if (user && user.username) {
    // Use user.username
} else {
    console.warn("User object is null or username is missing.");
}
`
Example Fix in Your Code: Assuming NotificationsBox.jsx is where the issue occurs:

`return (
    <div>
        {notifications && notifications.length > 0 ? (
            notifications.map(notification => (
                notification.user && notification.user.username ? (
                    <div key={notification.id}>
                        {notification.user.username}
                    </div>
                ) : (
                    <div key={notification.id || Math.random()}>Unknown User</div>
                )
            ))
        ) : (
            <p>No notifications</p>
        )}
    </div>
);
`
Please provide additional details if this does not work.
